### PR TITLE
Week 19 HP-API Done

### PR DESCRIPTION
**Pull request by Dopest0727**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,27 +1,96 @@
...
+// ROUTES
+app.get('/', (req, res) => {
+  res.send('Tjaba, this is my API for my frontend app, see it here "https://tiopunkter.netlify.app"')
+})
+
+// ENDPOIJNT SHOWS 20 LATEST MSG's
+app.get('/thoughts', async (req, res) => {
+  const thoughts = await Thought.find().sort({ createdAt: 'desc' }).limit(20).exec()
+  res.status(200).json(thoughts)
+})
```
Comment (Laura-Sjolander) [inline server.js:51]: I tried this endpoint in backend  Seems to work fine, All the messages/thoughts is there.

File: server.js [hunk 2]
```diff
@@ -1,27 +1,96 @@
...
+const Thought = mongoose.model('Thought', ThoughtSchema)
+
+// ADD MIDDLEWARE
+app.use(cors())
+app.use(express.json())
+
+// ROUTES
+app.get('/', (req, res) => {
+  res.send('Tjaba, this is my API for my frontend app, see it here "https://tiopunkter.netlify.app"')
+})
```
Comment (Laura-Sjolander) [inline server.js:45]: Tried this route, works well! Great!

File: server.js [hunk 3]
```diff
@@ -1,27 +1,96 @@
...
+  const updatedHearts = await Thought.findByIdAndUpdate(
+    thoughtId, 
+    { 
+      $inc: { 
+        hearts: 1
+      }
+    },
+    {
+      new: true
+    }
```
Comment (Laura-Sjolander) [inline server.js:78]: Nice way to update the hearts by incrementing

**Review by Laura-Sjolander — COMMENTED** [review]
Nice and clean code! Works well, I tried both the backend and the frontend, connection between them is there and stores messages as it should. 
And I love the design in the frontend part, as always. 
Keep on doing what you are doing!

File: server.js [hunk 4]
```diff
@@ -1,27 +1,96 @@
...
+  )
+
+  if (updatedHearts) {
+    res.json(updatedHearts)
+  } else {
+    res.status(404).json({ response: 'No happy thought with this ID', success: false })
```
Comment (jessand77) [inline server.js:84]: Good that you used the 404 status code here.

File: server.js [hunk 5]
```diff
@@ -1,27 +1,96 @@
...
+app.use(cors())
+app.use(express.json())
+
+// ROUTES
+app.get('/', (req, res) => {
+  res.send('Tjaba, this is my API for my frontend app, see it here "https://tiopunkter.netlify.app"')
+})
+
+// ENDPOIJNT SHOWS 20 LATEST MSG's
+app.get('/thoughts', async (req, res) => {
```
Comment (jessand77) [inline server.js:48]: A suggestion is to use a try and catch block here even if there is not so much that should be able to go wrong here :-)

**Review by jessand77 — COMMENTED** [review]
Good job with this week's project too :-) Everything seems to work as it should, the code has a good structure and includes error messages and validation.

**Pull request closed without merging**